google/cloud/aiplatform/training_jobs.py

```diff
@@ -261,7 +269,7 @@ def _create_input_data_config(
     def _run_job(
         self,
         training_task_definition: str,
-        training_task_inputs: dict,
+        training_task_inputs: proto.Message,
```

Comment:
For the custom jobs, I wrapped the dict like this before passing to `_run_job`. Are you okay with that?

```
json_format.ParseDict(
                training_task_inputs, struct_pb2.Value()
            )
```

Reply:
That works for me, but the other AutoML classes are passing in dicts as well: https://github.com/googleapis/python-aiplatform/blob/4f5797f65af03fc781c778e6f2d99c0f2d468920/google/cloud/aiplatform/training_jobs.py#L2643

So it may be cleaner and less code to accept Union[dict, proto.message] and conditionally create the struct_pb2.Value if passed a dict. Though, we may be able to bypass `json_format.ParseDict` and directly pass in the dict based on this: https://github.com/googleapis/proto-plus-python/blob/ccf903e3cddfcb1ff539e853594b4342914b7d61/docs/messages.rst#usage

Reply:
I imagined we would replace them with EPCL's at one point.

I will do this:
1. Change the docstring to allow a `Union of Dict + EPCL types`. 

I've also tested that passing in the `training_task_inputs_dict` as a dict WITHOUT conversion works fine.